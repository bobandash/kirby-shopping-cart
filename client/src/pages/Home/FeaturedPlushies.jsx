import styles from './FeaturedPlushies.module.css';
import sharedStyles from './shared.module.css';
import ClickableItem from './ClickableItem';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import dividerStyle from '../../components/divider.module.css'
import { useRef, useEffect, useState } from 'react';

function FeaturedPlushies({plushies}){
  const navigate = useNavigate();
  const headerRef = useRef(null);
  const plushBtnRef = useRef(null);
  const [isHeaderVisible, setIsHeaderVisible] = useState(false);
  const [isPlushBtnVisible, setIsPlushBtnVisible] = useState(false);
  useEffect(() => {
    const observer = new IntersectionObserver((entries) => {
      const headerEntry = entries[0];
      if(headerEntry.isIntersecting){
        setIsHeaderVisible(true);
      }

    });
    observer.observe(headerRef.current);

  }, []);

  useEffect(() => {
    const observer = new IntersectionObserver((entries) => {
      const plushBtnEntry = entries[0];
      if (plushBtnEntry.isIntersecting){
        setIsPlushBtnVisible(true);
      }    
    });
    observer.observe(plushBtnRef.current);
  }, []);


  function redirectPlushies(){
      navigate("/category/plushies")
  }
  

  const headerClasses = isHeaderVisible ? `${sharedStyles.header} ${sharedStyles.visible}` : sharedStyles.header;
  const plushBtnClasses = isPlushBtnVisible ? `${styles.visible} ${styles["shop-plushies-btn"]}` : styles["shop-plushies-btn"];
  return (
    <section className = {`${dividerStyle["divider"]} ${styles["featured-plush-container"]}`}>
      <div className = {`${sharedStyles.container} ${styles["align-container"]}`}>
        <h1 ref = {headerRef} className = {headerClasses}>Featured Plushies</h1>
        <div className = {styles["plush-container"]}>
          {plushies.map(plush => (
            <ClickableItem item = {plush} key = {plush.id}/>
          ))}
        </div>
        <button 
          ref = {plushBtnRef}
          onClick = {redirectPlushies}
          className = {plushBtnClasses}>
            Shop All Plushies
        </button>
      </div>
    </section>
  )
}

FeaturedPlushies.propTypes = {
  plushies: PropTypes.array,
}

export default FeaturedPlushies;