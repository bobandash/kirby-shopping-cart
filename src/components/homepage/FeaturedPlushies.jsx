import styles from './FeaturedPlushies.module.css';
import sharedStyles from './shared.module.css';
import ClickableItem from './ClickableItem';
import PropTypes from 'prop-types';

function FeaturedPlushies({plushies}){
  return (
    <section className = {styles["featured-plush-container"]}>
      <div className = {`${sharedStyles.container} ${styles["align-container"]}`}>
        <h1 className = {sharedStyles.header}>Featured Plushies</h1>
        <div className = {styles["plush-container"]}>
          {plushies.map(plush => (
            <ClickableItem plush = {plush} key = {plush.id}/>
          ))}
        </div>
        <button className = {styles["shop-plushies-btn"]}>Shop All Plushies</button>
      </div>
    </section>
  )
}

FeaturedPlushies.propTypes = {
  plushies: PropTypes.array,
}

export default FeaturedPlushies;