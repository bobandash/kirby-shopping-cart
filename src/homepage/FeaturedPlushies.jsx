import styles from './FeaturedPlushies.module.css';
import sharedStyles from './shared.module.css';
import ClickableItem from './ClickableItem';

function FeaturedPlushies(){
  return (
    <section className = {styles["featured-plush-container"]}>
      <div className = {`${sharedStyles.container} ${styles["align-container"]}`}>
        <h1 className = {sharedStyles.header}>Featured Plushies</h1>
        <div className = {styles["plush-container"]}>
          <ClickableItem />
          <ClickableItem />
          <ClickableItem />
        </div>
        <button className = {styles["shop-plushies-btn"]}>Shop All Plushies</button>
      </div>
    </section>
  )
}

export default FeaturedPlushies;