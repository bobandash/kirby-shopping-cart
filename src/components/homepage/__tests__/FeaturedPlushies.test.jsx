import FeaturedPlushies from "../FeaturedPlushies";
import {render, screen} from "@testing-library/react";
import { describe, it, expect } from 'vitest';

describe('Featured Plushies Component', () => {
  it('renders featured plushies component', () => {
    const { container } = render(<FeaturedPlushies />);
    expect( container ).toMatchSnapshot();
  })
})