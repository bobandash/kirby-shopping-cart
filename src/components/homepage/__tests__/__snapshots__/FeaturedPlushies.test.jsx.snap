// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Featured Plushies Component > renders featured plushies component 1`] = `
<div>
  <section
    class="_featured-plush-container_4adad7"
  >
    <div
      class="_container_59ef55 _align-container_4adad7"
    >
      <h1
        class="_header_59ef55"
      >
        Featured Plushies
      </h1>
      <div
        class="_plush-container_4adad7"
      >
        <div
          class="_card_fa9861"
        >
          <img
            alt="primary-image"
            src="/src/assets/product-assets/little_buddy_kirby_front.jpg"
          />
          <h2
            class="_product-name_fa9861"
          >
            Kirby
          </h2>
          <p
            class="_price_fa9861"
          >
            $19.99
          </p>
        </div>
        <div
          class="_card_fa9861"
        >
          <img
            alt="primary-image"
            src="/src/assets/product-assets/little_buddy_kirby_front.jpg"
          />
          <h2
            class="_product-name_fa9861"
          >
            Kirby
          </h2>
          <p
            class="_price_fa9861"
          >
            $19.99
          </p>
        </div>
        <div
          class="_card_fa9861"
        >
          <img
            alt="primary-image"
            src="/src/assets/product-assets/little_buddy_kirby_front.jpg"
          />
          <h2
            class="_product-name_fa9861"
          >
            Kirby
          </h2>
          <p
            class="_price_fa9861"
          >
            $19.99
          </p>
        </div>
      </div>
      <button
        class="_shop-plushies-btn_4adad7"
      >
        Shop All Plushies
      </button>
    </div>
  </section>
</div>
`;
